exec/jdbc/src/main/java/org/apache/drill/jdbc/impl/DrillConnectionImpl.java

```diff
@@ -362,12 +364,33 @@ public PreparedStatement prepareStatement(String sql, int resultSetType,
                                             int resultSetHoldability) throws SQLException {
     throwIfClosed();
     try {
-      DrillPrepareResult prepareResult = new DrillPrepareResult(sql);
+      DrillRpcFuture<CreatePreparedStatementResp> respFuture = client.createPreparedStatement(sql);
+
+      CreatePreparedStatementResp resp;
+      try {
+        resp = respFuture.get();
+      } catch (InterruptedException e) {
+        // Preserve evidence that the interruption occurred so that code higher up
+        // on the call stack can learn of the interruption and respond to it if it
+        // wants to.
+        Thread.currentThread().interrupt();
+
+        throw new SQLException( "Interrupted", e );
```

Comment:
What I was trying to understand was what happens if there is a message from the server on the wire and the respFuture gets interrupted. Now there is no listener to process the protobuf message. So who processes it?